libraries/AP_Compass/AP_Compass.h

```diff
@@ -333,10 +336,20 @@ friend class AP_Compass_Backend;
 
 private:
     static Compass *_singleton;
+
+    // Use Priority and StateIndex typesafe index types 
+    // to distinguish between two different type of indexing
+    // We use StateIndex for access by Backend
+    // and Priority for access by Frontend
+    DECLARE_TYPESAFE_INDEX(Priority, int);
```

Comment:
We don't normally use "int" but instead "uint8_t" or "uint16_t"

Reply:
will do